OpenRA.Mods.Common/Widgets/Logic/ReplayUtils.cs

```diff
@@ -0,0 +1,58 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using OpenRA.FileFormats;
+
+namespace OpenRA.Mods.Common.Widgets.Logic
+{
+	public static class ReplayUtils
+	{
+		static readonly Action DoNothing = () => { };
+
+		public static bool CheckReplayCompatibility(ReplayMetadata replayMeta, Action onCancel = null)
+		{
+			if (onCancel == null)
+				onCancel = DoNothing;
+
+			if (replayMeta == null)
+				return IncompatibleReplayDialog("an outdated engine", null, onCancel);
+
+			var version = replayMeta.GameInfo.Version;
+			if (version == null)
+				return IncompatibleReplayDialog("an unknown version", version, onCancel);
+
+			var mod = replayMeta.GameInfo.Mod;
+			if (mod == null)
+				return IncompatibleReplayDialog("an unknown mod", mod, onCancel);
+
+			var allMods = ModMetadata.AllMods;
+			if (!allMods.ContainsKey(mod))
+				return IncompatibleReplayDialog("an unavailable mod", mod, onCancel);
+			else if (allMods[mod].Version != version)
+				return IncompatibleReplayDialog("an incompatible version", version, onCancel);
+
+			if (replayMeta.GameInfo.MapPreview.Status != MapStatus.Available)
+				return IncompatibleReplayDialog("an unavailable map", replayMeta.GameInfo.MapUid,  onCancel);
+			else
+				return true;
+		}
+
+		static bool IncompatibleReplayDialog(string type, string name, Action onCancel)
+		{
+			var error = "It was recorded with " + type;
```

Comment:
This is pretty nit-picky of me but since you are really hardcoding the different reasons here this string may as well include the "an" that all of the reasons start with, instead of duplicating it.